src/providers/NetworkProvider.tsx

```diff
@@ -19,9 +16,7 @@ export const NetworkProvider: React.FC = ({ children }) => {
   const { isDarkMode } = useContext(ThemeContext)
 
```

Comment:
some general reorganization here in addition to supporting new NetworkProvider properties.

main idea is replacing as much logic as possible with onboard subscriptions